plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2071,6 +2120,14 @@ protected StartAnswer execute(StartCommand cmd) {
         }
     }
 
+    private void replaceNicsMacSequenceInBootArgs(String oldMacSequence, String newMacSequence, VirtualMachineTO vmSpec) {
+        String bootArgs = vmSpec.getBootArgs();
+        if (!StringUtils.isEmpty(bootArgs) && !StringUtils.isEmpty(oldMacSequence) && !StringUtils.isEmpty(newMacSequence)) {
+            //Update boot args with the new nic mac addresses
```

Comment:
@rafaelweingartner Moved the comment to the method documentation part. Testcase added.